oci/config/convert/runtime.go

```diff
@@ -124,30 +128,26 @@ func MutateRuntimeSpec(g rgen.Generator, rootfs string, image ispec.Image, manif
 	g.SetProcessUID(uint32(execUser.Uid))
 	g.SetProcessGID(uint32(execUser.Gid))
 	g.ClearProcessAdditionalGids()
+
 	for _, gid := range execUser.Sgids {
 		g.AddProcessAdditionalGid(uint32(gid))
 	}
 	if execUser.Home != "" {
 		g.AddProcessEnv("HOME", execUser.Home)
 	}
 
+	// Set optional fields
+	var ports []string
+	for port := range image.Config.ExposedPorts {
+		ports = append(ports, port)
+	}
```

Comment:
As we discussed you'll need to sort `ports` after this because iteration over Go maps is random. You can use the `sort` package with `sort.StringSlice` (https://golang.org/pkg/sort/#StringSlice).

Though in future it would be nice if this was hidden behind the `github.com/openSUSE/umoci/oci/config/generate` API.